### ISSUE TITLE
Add Ajax csrf_token to base.html

### ISSUE THREAD

**Issue by mmmcorpsvit** [pydanny/cookiecutter-django#2207]
Add this code to base.html

```html
  <script>
    $.ajaxSetup({
      data: {csrfmiddlewaretoken: '{{ csrf_token }}'},
    });
  </script>
```
this allow easy use Ajax forms + csrf token

**Comment by mmmcorpsvit** [comment]
can help too: https://stackoverflow.com/questions/5100539/django-csrf-check-failing-with-an-ajax-post-request

**Comment by demestav** [comment]
Quoting from the [documentation](https://api.jquery.com/jQuery.ajaxSetup/)

> All subsequent Ajax calls using any function will use the new settings, unless overridden by the individual calls, until the next invocation of `$.ajaxSetup()`.

If I am reading this correctly, if you make any ajax call that will contain request data (i.e. overriding the `data` parameter), and this is highly likely, then you would need to specify the csrf token again. Therefore, there is little benefit to add it by default.

**Comment by mmmcorpsvit** [comment]
no, this add csrfmiddlewaretoken token to data dict, so if you already use JQuery and use data usual this not must be problem)

**Comment by davitovmasyan** [comment]
@demestav agreed with you. 

However, there is a better solution for CSRF and jQuery ajax, which is also mentioned in [django's docs](https://docs.djangoproject.com/en/2.2/ref/csrf/#acquiring-the-token-if-csrf-use-sessions-or-csrf-cookie-httponly-is-true). 

If you think that having an `X-CSRFToken` header with a `csrftoken` value in every ajax call<sup>1</sup> will give some benefits to this project, then the suggestion from Django's docs could be a solution for this issue.

<sup>1</sup> except those((GET|HEAD|OPTIONS|TRACE)) which doesn't require CSRF protection
